dhall/src/Dhall/Parser/Expression.hs

```diff
@@ -443,18 +444,29 @@ parsers embedded = Parsers {..}
     selectorExpression = noted (do
             a <- primitiveExpression
 
-            let recordType = _openParens *> whitespace *> expression <* whitespace <* _closeParens
+            let recordType = whitespace *> _openParens *> whitespace *> expression <* whitespace <* _closeParens
 
             let field               x  e = Field   e  x
             let projectBySet        xs e = Project e (Left  xs)
             let projectByExpression xs e = Project e (Right xs)
 
+            let fieldParser = do
+                    src0 <- src whitespace
+                    l <- anyLabel
+                    pos <- getSourcePos
+
+                    -- FIXME: Suffix whitespace can't be parsed given our limitation
+                    -- about whitespace treatment, but for @dhall-docs@ this
+                    -- is enough
+                    let src1 = Src pos pos ""
+                    return (FieldAccess (Just src0) l (Just src1))
```

Comment:
I don't understand this FIXME.  `anyLabel` doesn't eat trailing whitespace, or does it?!

I think `src1` should properly include all the whitespace until the next `.`.

I also think we direly need tests for the whitespace parsing – possibly golden tests based on `haskell-syntax-tree`?! That's for future work though, I guess.

Reply:
I had a lot of parsing errors from our test-suite when I added the `src1 <- src whitespace` there, I can try to reproduce and paste a sample of the failure I got to get more help from you and @Gabriel439

I agree with the test-setup you suggest, I'll file an issue to not forget about it

Reply:
Alright, after replacing the `let src1 = Src pos pos ""` with `src1 <- src whitespace` I got a lot of parse errors from the test-suite. This is a single test failure:

```
Dhall Tests
  normalization
    discover
      ./dhall-lang/tests/normalization/success/remoteSystems:                                                                  FAIL
        Exception: 
        [1;31mError[0m: Invalid input
        
        41:3:
           |
        41 |   (   λ ( x
           |   ^
        unexpected '('
        expecting ->, :, end of input, or whitespace
```

the error comes from parsing https://github.com/dhall-lang/dhall-lang/blob/master/tests/normalization/success/remoteSystemsB.dhall

do you have any idea? @Gabriel439 may have a solution or more insights about the reason for the failure.

Reply:
Thanks for investigating, @german1608! I'm fine with leaving this bit as it is for now then.

Reply:
You're welcome! I think that I'll update #1995 to not forget about this